libs/math/include/math/TVecHelpers.h

```diff
@@ -106,6 +106,22 @@ class TVecAddOperators {
         return operator-=(VECTOR<U>(v));
     }
 
+    constexpr VECTOR<T>& operator+=(T s) {
```

Comment:
mh... no I don't like that. This should not be needed at all, since we already have a template just above that should handle this. Any idea why this is not working with MSVC?

in this case:
```
double d = ...;
float3 u = ...;
auto r = u * d;  // r should be a double3
```

I'm worried that it won't see the template above and use this one instead, which would down cast d to float.

Reply:
You are right, these are not needed. At one point I think I made the TVec3(float) constructor explicit, so these were needed, but not anymore. I removed them.